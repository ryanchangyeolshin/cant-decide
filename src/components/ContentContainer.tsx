
import React, { Fragment, useContext, useState } from 'react'
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { Typography, List, CssBaseline } from '@material-ui/core';
import ChoiceCardContainer from './ChoiceCardContainer';
import InputContainer from './InputContainer';
import DecisionContext from '../context/DecisionContext';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    containerWrapper: {
      maxWidth: "100%"
    },
    contentContainer: {
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
      backgroundColor: "#4051B6",
      height: "100vh",
      margin: "50px 50px",
      borderRadius: "20px",
    },
    list: {
      width: '100%',
      borderRadius: "4px",
      height: "78%",
      overflow: "scroll",
      paddingLeft: "20px",
      paddingRight: "20px"
    },
  }),
);

const ContentContainer = () => {
  const classes = useStyles();
  const { setWinningChoice, decision, setDecision } = useContext(DecisionContext);
  const [choice, setChoice] = useState<string>("");

  const handleChoiceChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setChoice(e.target.value);
  };

  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.key === 'Enter' && choice !== "") {
      setDecision({ ...decision, choices: decision.choices.concat(choice) });
      setChoice("");
    }
  }

  const removeChoice = (index: number) => {
    const filteredChoices: string[] = decision.choices.filter((choice: string, i: number) => i !== index);
    setDecision({ ...decision, choices: filteredChoices });
  }

  const randomizeChoice = (e: React.MouseEvent<HTMLButtonElement>) => {
    const randomIndex: number = Math.floor(Math.random() * decision.choices.length);
    setWinningChoice(decision.choices[randomIndex]);
  }

  return (
    <Fragment>
      <div className={classes.containerWrapper}>
        <CssBaseline />
        <Typography component="div" className={classes.contentContainer}>
          <InputContainer
            choice={choice}
            handleKeyDown={handleKeyDown}
            handleChoiceChange={handleChoiceChange}
            randomizeChoice={randomizeChoice}
          />
          <div className={classes.list}>
            <List component="nav" aria-label="main mailbox folders">
              {decision.choices.map((newChoice: string, i: number) =>
                <ChoiceCardContainer
                  key={i}
                  choice={newChoice}
                  index={i}
                  removeChoice={removeChoice}
                />
              )}
            </List>
          </div>
        </Typography>
      </div>
    </Fragment>
  );
};

export default ContentContainer;