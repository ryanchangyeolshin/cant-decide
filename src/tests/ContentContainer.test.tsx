import React from 'react';
import { render } from '@testing-library/react';
import ContentContainer from '../components/ContentContainer';

describe("This suit is to test the ContentContainer component", () => {
  test('Snapshot of ContentContainer', () => {
    const { asFragment } = render(<ContentContainer />);
    expect(asFragment()).toMatchSnapshot()
  });
});