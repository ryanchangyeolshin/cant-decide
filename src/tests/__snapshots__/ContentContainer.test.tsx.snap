// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the ContentContainer component Snapshot of ContentContainer 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-containerWrapper-1"
  >
    <div
      class="MuiTypography-root makeStyles-contentContainer-2 MuiTypography-body1"
    >
      <div
        class="makeStyles-inputContainer-4"
      >
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-choiceInput-5 MuiFormControl-marginNormal"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-7 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="standard-basic"
            id="standard-basic-label"
          >
            Choice
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input makeStyles-input-6"
              id="standard-basic"
              placeholder="Please provide your choice here..."
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-11"
              >
                <span>
                  Choice
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-randomize-8 MuiButton-containedPrimary Mui-disabled Mui-disabled"
          data-testid="randomize-button"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Randomize
          </span>
        </button>
      </div>
      <div
        class="makeStyles-list-3"
      >
        <nav
          aria-label="main mailbox folders"
          class="MuiList-root MuiList-padding"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
